src/test/java/com/microsoft/sqlserver/jdbc/TestUtils.java

```diff
@@ -666,4 +668,34 @@ public static boolean supportJDBC43(Connection con) throws SQLException {
     public static String escapeSingleQuotes(String name) {
         return name.replace("'", "''");
     }
+    
+    /**
+     * Returns if connected to SQL Azure
+     * @param con
+     *        connection to server
+     * @return boolean
+     * @throws SQLException
+     */
+    public static boolean isSqlAzure(Connection con) throws SQLException {
+        try (ResultSet rs = con.createStatement().executeQuery("SELECT CAST(SERVERPROPERTY('EngineEdition') as INT)")) {
+            rs.next();
+            int engineEdition = rs.getInt(1);
+            return (engineEdition == ENGINE_EDITION_FOR_SQL_AZURE || engineEdition == ENGINE_EDITION_FOR_SQL_AZURE_DW);
```

Comment:
Can you use a static variable (like how it's done in our fx framework) to mark if the server's type has been determined and re-use that, so that we don't have to contact the server more than once?